### PR TITLE
[ASTS] Telemetry part 1: Single bucket sweep task

### PR DESCRIPTION
**Pull request by mdaudali**
## General
**Before this PR**:
We don't actually have any telemetry for what happened during a single bucket sweep task, which will make it hard to debug when we're testing in the wild.
**After this PR**:
Wires up the existing sweep outcome telemetry

==COMMIT_MSG==
==COMMIT_MSG==

**Priority**: P2

**Concerns / possible downsides (what feedback would you like?)**:
This is not the final design. I do later want the sweep task to report a status back to the coordinator to centralise metrics and for the autoscaler, but I don't know what the autoscaling work will look like yet.
**Is documentation needed?**:

## Compatibility
N/A

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:
N/A
**What was existing testing like? What have you done to improve it?**:
Updated tests
## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:
Metrics are emitted
**Has the safety of all log arguments been decided correctly?**:
Yes
**Will this change significantly affect our spending on metrics or logs?**:
No
**How would I tell that this PR does not work in production? (monitors, etc.)**:
Metrics are not emitted but the age of latest swept entry monitor fires
**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:
Turn off ASTS
**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:
N/A

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@raiju

### CODE REVIEW

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java [hunk 1]
```diff
@@ -234,9 +244,62 @@ public void sweepsEmptyNonDeleteableBucketEntirelyAndDoesNotDeleteBucket(SweepBu
         verify(bucketProgressStore)
                 .updateBucketProgressToAtLeast(
                         context.bucket(), context.completeProgressForBucket().orElseThrow());
+        verify(targetedSweepMetrics).registerOccurrenceOf(context.shardAndStrategy(), SweepOutcome.NOTHING_TO_SWEEP);
         verifyNoInteractions(completionListener);
     }
 
+    @ParameterizedTest
+    @MethodSource("closedSweepBuckets")
+    public void recordsNotEnoughDbNodesOnlineWhenFailingToSweepBucketDueToInconsistentConsistency(
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java]: nit: insufficient

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTask.java [hunk 2]
```diff
@@ -69,9 +71,14 @@ public long runOneIteration(SweepableBucket sweepableBucket) {
...
                 Sweeper.of(sweepableBucket.bucket().shardAndStrategy()).getSweepTimestamp(specialTimestampsSupplier);
         long bucketStartTimestamp = sweepableBucket.timestampRange().startInclusive();
+
+        // TODO(mdaudali): we do need a way to report back to the coordinator about what happened when we tried to
+        //  sweep, so that the autoscaler can adjust as necessary and we can centralise metric reporting.
         if (sweepTimestampForIteration <= bucketStartTimestamp) {
             // This means that the sweep timestamp has not entered this partition yet, so we do not need to process
             // anything. Note that sweep timestamps are exclusive, so <= is correct.
+            targetedSweepMetrics.registerOccurrenceOf(
+                    sweepableBucket.bucket().shardAndStrategy(), SweepOutcome.NOTHING_TO_SWEEP);
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTask.java:81]: I guess nowadays we already do it for the shard, though one of the things I didn't like when thinking about this one was that it's more _this specific bucket_ having nothing to sweep, and so the reader of this code needs to bear that in mind.

**Review by jeremyk-91 — APPROVED** [review]
👍

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java (hunk 1 above)
Comment (mdaudali) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java]:
```suggestion
    public void recordsNotEnoughDbNodesOnlineWhenFailingToSweepBucketDueToInsufficientConsistency(
```

**Pull request merged**